db/users.go

```diff
@@ -77,7 +79,7 @@ func (dbc *DatabaseContext) UpdatePrincipal(newInfo PrincipalConfig, isUser bool
 	var user auth.User
 	authenticator := dbc.Authenticator()
 	if isUser {
-		isValid, reason := newInfo.IsPasswordValid(dbc.AllowEmptyPassword)
+		isValid, reason := newInfo.IsPasswordValid(dbc.AllowEmptyPassword || allowReplace)
```

Comment:
Yeah, the 'request != POST' in particular was not working as intended, I have reverted this and take a different approach.